lib/rex/post/meterpreter/extensions/stdapi/railgun/def/windows/def_advapi32.rb

```diff
@@ -1263,12 +1263,12 @@ def self.create_library(constant_manager, library_path = 'advapi32')
 
     dll.add_function('ConvertStringSidToSidA', 'BOOL',[
       ["PCHAR","StringSid","in"],
-      ["PDWORD","pSid","out"],
+      ["PLPVOID","pSid","out"],
```

Comment:
This should be an LPVOID shouldn't it? It's a pointer to data not a pointer to a pointer like PLPVOID describes.

Alternatively, this should probably be a PBLOB so you could pass it a packed value.

It's definitely not a pointer to a pointer though.

Reply:
So the part I was confused about was the fact that according to https://sourcegraph.com/github.com/rapid7/metasploit-framework/-/blob/lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb?L48-49 and https://sourcegraph.com/github.com/rapid7/metasploit-framework/-/blob/lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb?L226 which states that `LPVOID` is special in that like `PBLOB` its not backed by railgun memory.

I also notice that https://sourcegraph.com/github.com/rapid7/metasploit-framework/-/blob/lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb?L155 notes that LPVOID is not actually treated as a pointer.

Would appreciate some clarity on this given this information.

Reply:
Also now that I look at it this field is defined as `[out] PSID   *Sid` and as `A pointer to a variable that receives a pointer to the converted SID. To free the returned buffer, call the [LocalFree](https://learn.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-localfree) function.` so I think it is actually a pointer to a pointer?

Reply:
Ah you're right, I was getting this confused with the other function we were looking at. `PSID *pSid` would be a pointer to a pointer.